src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -341,6 +341,13 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                 _pageContext.ViewData.Model = _model;
             }
 
+            if (CacheEntry.PropertyBinder != null &&
+                !string.Equals(_pageContext.HttpContext.Request.Method, "GET", StringComparison.OrdinalIgnoreCase))
```

Comment:
Totally stray thought: It'd be nice to have extension methods somewhere in the stack like `_pageContext.HttpContext.Request.IsGet()`